pkg/node/manager/rest_api.go

```diff
@@ -166,7 +167,7 @@ func (c *clusterNodesClient) NodeDelete(node nodeTypes.Node) error {
 		}
 	}
 	if found != -1 {
-		c.NodesAdded = append(c.NodesAdded[:found], c.NodesAdded[found+1:]...)
+		c.NodesAdded = slices.Delete(c.NodesAdded, found, found+1)
```

Comment:
Nit. The hunk immediately above could be replaced by `slices.IndexFunc` (or just calling `DeleteFunc`, but that then requires tracking whether an element got deleted to still handle the else branch).

Reply:
Nice catch. I think using `slices.IndexFunc` and a successive `slices.Delete` makes this slightly more readable. Changed accordingly.